internal/adapters/runtime/kubernetes/game_room_convert.go

```diff
@@ -61,10 +62,12 @@ var invalidPodWaitingStates = []string{
 }
 
 func convertGameRoomSpec(schedulerID string, gameRoomSpec game_room.Spec) (*v1.Pod, error) {
+	randID := strings.SplitN(uuid.NewV4().String(), "-", 2)[0]
```

Comment:
inspired by https://github.com/topfreegames/maestro/blob/master/models/game_room.go#L158

Reply:
We should be careful with the name of the pods we generate (that's why we were delegating to Kubernetes to do it).

WDYT about using the same logic the `GenerateName`? We would be just changing to our side so that we can place our environment variable. Ref: https://github.com/kubernetes/apiserver/blob/master/pkg/storage/names/generate.go#L40

Reply:
Alternatively, we can add an environment variable that references the Pod resource field. An example of this in YAML format would be:
```yaml
- name: MAESTRO_ROOM_ID
  valueFrom:
    fieldRef:
      fieldPath: status.podIP
```

Ref: https://pkg.go.dev/k8s.io/api/core/v1#EnvVarSource

Reply:
3c1b0a8770b93a11143f0d240fa8153e58e31262